src/to-do.js

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable quotes */
+const InnerTodo = document.querySelector(".todo-body");
+let isEdit = false;
+let editId = null;
+
+export default class display {
+  static getTodo = () => {
+    let Todo;
+    if (localStorage.getItem("todo") == null) {
+      Todo = [];
+    } else {
+      Todo = JSON.parse(localStorage.getItem("todo"));
+    }
+    return Todo;
+  };
+
+  static addRemoveEvent = () => {
+    const trash = document.querySelectorAll(".trash");
+    trash.forEach((task, i) => {
+      task.addEventListener("click", (ev) => {
+        ev.preventDefault();
+        this.removeTodo(i);
+      });
+    });
+  };
+
+  static removeTodo = (id) => {
+    const toDos = display.getTodo();
+    toDos.splice(id, 1);
+    toDos.forEach((item) => {
+      if (item.index > id) {
+        item.index -= 1;
+      }
+    });
+    localStorage.setItem("todo", JSON.stringify(toDos));
+    this.loadTodo(toDos);
+  };
+
+  static loadTodo = (item) => {
+    let display = " ";
+    item.forEach((elem, i) => {
+      display += `
+      <div class="todo-check flex">
+      <div class="checkbox">
+        <input
+          type="checkbox"
+          id="to-do-check"
+          name="To-Do"
+          value="Add" maxlength="10"/>
+        <label for="todo">${elem.text}</label><br />
+      </div>
+      <div class= "check-icons">
+      <div class="trash">
+      <i class="fa-solid fa-trash" id="${i}"></i>
+      </div>
+      <i class="edit-btn vertical-menu fa-solid fa-ellipsis-vertical" id="${i}"></i>
+      </div>
+    </div>
+    <hr />`;
+    });
+    InnerTodo.innerHTML = display;
+    this.addRemoveEvent();
+    this.editEvent();
+  };
+
+  static addTodo = () => {
+    const text = document.querySelector(".type-task").value;
+    if (text !== "") {
+      const toDos = display.getTodo();
+      const newInput = { text, completed: false, index: toDos.length };
+      const editInput = { text, completed: false, index: editId };
+
+      if (isEdit) {
+        const singleTodo = toDos.find((item, index) => index === editId);
+        Object.assign(singleTodo, editInput);
+        localStorage.setItem("todo", JSON.stringify(toDos));
+        this.loadTodo(toDos);
+        isEdit = false;
+        editId = null;
+        document.querySelector(".type-task").value = "";
+        return;
+      }
+      toDos.push(newInput);
+      localStorage.setItem("todo", JSON.stringify(toDos));
+      this.loadTodo(toDos);
+      document.querySelector(".type-task").value = "";
+    }
+  };
```

Comment:
- [ ] The `index` property of the tasks should reflect their position in the todo array. That means the first task should have an index of 1 (not 0), the second one should have an index of 2 (not 1), and so on. Kindly fix that.
  
  ![ifofr](https://user-images.githubusercontent.com/81468675/213489233-37b6f070-826d-4612-a520-76fc905d5b9e.jpg)

  This should be something like this:
  
  ```
  todo:
    [
      {"text":"First task","completed":false,"index":1},
      {"text":"Second task","completed":false,"index":2},
      {"text":"Third task","completed":false,"index":3}
    ]
  ```